public/controllers/agent/components/register-agent.js

```diff
@@ -523,17 +531,25 @@ export const RegisterAgent = withErrorBoundary(
                   iconType="iInCircle"
                 />
                 <EuiSpacer />
-                <EuiCodeBlock style={codeBlock} language={language}>
-                  {this.state.wazuhPassword ? this.obfuscatePassword(text) : text}
-                </EuiCodeBlock>
+                <div className="copy-codeblock-wrapper">
+                  <EuiCodeBlock style={codeBlock} language={language}>
+                    {this.state.wazuhPassword && !this.state.showPassword ? this.obfuscatePassword(text) : text}
+                  </EuiCodeBlock>
+                  <EuiCopy textToCopy={text}>
+                    {(copy) => (
+                      <div className="copy-overlay"  onClick={copy}>
+                        <p><EuiIcon type="copy"/> Copy command</p>
+                      </div>
+                    )}
+                  </EuiCopy>
+                </div>
+                <EuiSwitch
+                  label="Show password"
+                  checked={this.state.showPassword}
+                  onChange={(active) => this.setShowPassword(active)}
+                />
+                <EuiSpacer />
```

Comment:
**nitpick**: The switch that manages the visibility of the password should only be displayed when there is one.

```jsx
{this.state.wazuhPassword && (
  <EuiSwitch
    label="Show password"
    checked={this.state.showPassword}
    onChange={(active) => this.setShowPassword(active)}
  />
                <EuiSpacer />
)}
```

Reply:
great suggestion